src/components/button/index.tsx

```diff
@@ -1,16 +1,32 @@
 import React, { ReactNode } from 'react';
 import { WrapperButton, TextButton } from './styles';
+import { ActivityIndicator } from 'react-native';
 
+type ButtonVariant = 'BTN_DETAILS' | 'BTN_PRIMARY' | 'BTN_ALERT' | 'BTN_SUCCESS';
 interface ButtonProps {
-  children?: ReactNode;
   onPress?: () => void;
   disabled?: boolean;
+  text: string;
+  width: string;
+  variant: ButtonVariant;
+  icon?: ReactNode;
+  onLoading?: boolean;
 }
 
-export function Button({ children, onPress }: ButtonProps) {
+export function Button({ onPress, text, width, icon, onLoading, ...props }: ButtonProps) {
   return (
-    <WrapperButton onPress={onPress}>
-      <TextButton>{children}</TextButton>
-    </WrapperButton>
+    <>
+      {onLoading === true ? (
+        <WrapperButton width={width} onPress={onPress} disabled={onLoading === true} {...props}>
+          <ActivityIndicator />
+          <TextButton>Carregando...</TextButton>
+        </WrapperButton>
+      ) : (
+        <WrapperButton width={width} onPress={onPress} {...props}>
+          {icon}
+          <TextButton>{text}</TextButton>
+        </WrapperButton>
+      )}
```

Comment:
ta usando o `onPress` e o `width` pra alguma outra coisa além de passar pro wrapper? se não daria pra parras no próprio `{...props}`

```suggestion
    <WrapperButton width={width} onPress={onPress} disabled={onLoading} {...props}>
      {onLoading ? (
          <ActivityIndicator />
          <TextButton>Carregando...</TextButton>
      ) : (
          {icon}
          <TextButton>{text}</TextButton>
      )}
    </WrapperButton>
```